features/learn/src/main/kotlin/com/teacherworkout/features/learn/discover/DiscoverScreen.kt

```diff
@@ -0,0 +1,194 @@
+                .fillMaxSize()
+                .padding(horizontal = space16dp),
+            verticalArrangement = Arrangement.spacedBy(space16dp),
+            state = listState
+        ) {
+            item {
+                Spacer(modifier = Modifier.height(space16dp))
+                Text(
+                    text = stringResource(id = R.string.discover_lessons_title),
+                    style = TextStyle(
+                        color = MaterialTheme.colors.primary,
+                        fontSize = MaterialTheme.typography.h4.fontSize,
+                        fontWeight = FontWeight.Bold
+                    )
+                )
+                Spacer(modifier = Modifier.height(space8dp))
+                SearchView(
+                    modifier = Modifier.fillMaxWidth(),
+                    searchInput = state.searchInput,
+                    onSearchInputChange = {
+                        onSendEvent(DiscoverContract.Event.SetSearchInput(it))
+                    },
+                    onClear = {
+                        onSendEvent(DiscoverContract.Event.SetSearchInput(TextFieldValue("")))
+                    },
+                    placeholderText = stringResource(id = R.string.search_label)
+                )
+                Spacer(modifier = Modifier.height(space16dp))
+            }
+            if(state.isLoading) {
+                item {
+                    Box(
+                        //TODO: look for a better way to center the progress bar
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .fillParentMaxHeight(0.5f),
+                        contentAlignment = Alignment.Center
+                    ) {
+                        CircularProgressIndicator()
+                    }
+                }
+            } else {
+                LessonThemes(
+                    lessonThemes = state.lessonThemes,
+                    onLessonThemeClick = { lessonTheme ->
+                        onSendEvent(DiscoverContract.Event.SelectLessonTheme(lessonTheme.title))
+                    }
+                )
+                item {
+                    Spacer(modifier = Modifier.height(space16dp))
+                }
+            }
+        }
+        SnackbarHost(
+            modifier = Modifier.align(Alignment.BottomCenter),
+            hostState = snackbarHostState
+        )
+    }
+}
+
+//TODO: reuse this in `HomeScreen`
+fun LazyListScope.LessonThemes(
+    lessonThemes: List<LessonTheme>,
+    onLessonThemeClick: (LessonTheme) -> Unit
+) {
+    item {
+        Text(
+            text = stringResource(id = R.string.themes_title),
+            style = TextStyle(
+                color = MaterialTheme.colors.primary,
+                fontSize = MaterialTheme.typography.h5.fontSize,
+                fontWeight = FontWeight.Bold
+            )
+        )
+    }
+
+    items(lessonThemes.chunked(2)) { oneOrTwoThemes ->
```

Comment:
That `2` shouldn't be changed. If I were to make it a variable, that means that I could safely write a 3 in there instead of 2. Correct me if I'm wrong.

Reply:
Yeah, totally right, was only thinking in terms of making the code easier to read. 

e.g: items(lessonThemes.chunked(NumerOfItemPerRow))
 instead of items(lessonThemes.chunked(2))
 
 Pretty sure you can think of a better name in case we want to do this refactoring.

Reply:
I'd say leave it like this for the moment.. If we were to change that `2` to a variable, we should definitely change the `oneOrTwoThemes` name and also the method's implementation because that is fully dependent on the value being `2`. I've tried to come up with some refactoring but I'm kind of overengineering it atm.